### PR TITLE
Feature/sof 5238

### PR DESCRIPTION
**Pull request by adewyer**

### CODE REVIEW

File: src/material.js [hunk 1]
```diff
@@ -144,6 +150,10 @@ export class Material {
         return this.prop('basis', undefined, true);
     }
 
+    set basis(config) {
```
Comment (timurbazhirov) [inline src/material.js:153]: There's `setBasis` already above

Comment (adewyer) [reply src/material.js:153]: There is but it takes in 3 parameters. I want a function that will take in a basis that already exists and apply it a material.

The other function seems more appropriate if you're trying to update based on content from a text file or from changes in the basis editor.

File: src/material.js [hunk 2]
```diff
@@ -158,12 +162,26 @@ export class Material {
...
+            coordinates: config.coordinates,
+            units: config.units,
+            cell: config.cell,
+            isEmpty: false
+        };
+        return new ConstrainedBasis({
+                ...newConfig
+            }
+        );
+    }
```
Comment (timurbazhirov) [inline src/material.js:177]: A setter that returns a Class instance???

File: src/material.js (hunk 1 above)
Comment (timurbazhirov) [reply src/material.js:153]: OK

File: src/material.js (hunk 2 above)
Comment (timurbazhirov) [reply src/material.js:177]: Let's remove this pls

File: src/material.js [hunk 3]
```diff
@@ -162,20 +164,6 @@ export class Material {
...
-            units: config.units,
-            cell: config.cell,
-            isEmpty: false
-        };
-        return new ConstrainedBasis({
-                ...newConfig
-            }
-        );
-    }
-
```
Comment (adewyer) [inline src/material.js]: This function ended up not being needed after updating the function `convertMDMaterialToMaterial(material)` in the web-app (https://github.com/Exabyte-io/exabyte-web-app/pull/1663/files#)

File: src/tools/material.js [hunk 4]
```diff
@@ -39,7 +40,28 @@ function scaleNonPeriodicLattice(material) {
     return newLattice;
 }
 
+function getBasisConfigTranslatedToCenter(material) {
```
Comment (timurbazhirov) [inline src/tools/material.js:43]: The idea is that the tools are working on a material object and return a modified material. We can extract the config by running `toJSON` on `material.Basis`

File: src/tools/material.js [hunk 5]
```diff
@@ -32,7 +34,26 @@ function scaleLatticeToMakeNonPeriodic(material) {
...
+ * so that the center of the material and lattice are aligned.
+ * @param material {Material}
+ * */
+function getBasisConfigTranslatedToCenter(material) {
+    const lattice = new Lattice(material.lattice);
+    const basis = new Basis({
+        ...material.Basis.toJSON()
+    });
+    basis.toCartesian();
+    const centerOfCoordinates = basis.centerOfCoordinatesPoint;
```
Comment (timurbazhirov) [inline src/tools/material.js:48]: Seems like too much code for what it needs to be

Why not just work on the `material.Basis` directly? There's also `material.toCartesian` function used above in the file...

**Pull request merged**